core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -1484,6 +1483,8 @@ public void up(MessageBatch batch) {
 
             nonBlockingExecutor.execute(() -> processMessage(message));
          });
+
+         processMessage(firstMessage.get());
```

Comment:
looking at JGroups code, `message` can be `null` in `forEach()´. it is probably worth checking for `null`.

Reply:
I didn't add a check here, as I'm assuming each batch must have at least one non-null message, but I added a check at the beginning of the lambda (line 1472)